src/Connection/Http1Connection.php

```diff
@@ -156,7 +154,6 @@ public function getStream(Request $request): Promise
 
     private function free(): Promise
     {
-        $this->socket = null;
```

Comment:
We have several null checks now, so this should be unnecessary. Or should we revert the null checks?

Reply:
I think this was a rebase mistake. I didn't mean to remove that.